src/harvest-rpde.js

```diff
@@ -0,0 +1,205 @@
+  config: { WAIT_FOR_HARVEST, VALIDATE_ONLY, VERBOSE, ORDER_PROPOSALS_FEED_IDENTIFIER, REQUEST_LOGGING_ENABLED } = {
+    WAIT_FOR_HARVEST: true, VALIDATE_ONLY: false, VERBOSE: false, ORDER_PROPOSALS_FEED_IDENTIFIER: null, REQUEST_LOGGING_ENABLED: false,
+  },
+  options: { multibar, pauseResume } = { multibar: null, pauseResume: null },
+}) {
+  let isInitialHarvestComplete = false;
+  let numberOfRetries = 0;
+  if (!context) context = createFeedContext(feedContextIdentifier, baseUrl, multibar);
+
+  if (feedContextMap.has(feedContextIdentifier)) {
+    throw new Error('Duplicate feed identifier not permitted within dataset distribution.');
+  }
+  feedContextMap.set(feedContextIdentifier, context);
+  let url = baseUrl;
+
+  // One instance of FeedPageChecker per feed, as it maintains state relating to the feed
+  const feedChecker = new FeedPageChecker();
+
+  // Harvest forever, until a 404 is encountered
+  for (; ;) {
+    // If harvesting is paused, block using the mutex
+    if (pauseResume) await pauseResume.waitIfPaused();
+
+    try {
+      const axiosConfig = {
+        headers: await headers(),
+      };
+
+      const timerStart = performance.now();
+      const response = await axios.get(url, axiosConfig);
+      const timerEnd = performance.now();
+      const responseTime = timerEnd - timerStart;
+
+      const json = response.data;
+
+      // Validate RPDE page using RPDE Validator, noting that for non-2xx state.pendingGetOpportunityResponses that we want to retry axios will have already thrown an error above
+      const rpdeValidationErrors = feedChecker.validateRpdePage({
+        url,
+        json,
+        pageIndex: context.pages,
+        contentType: response.headers['content-type'],
+        cacheControl: response.headers['cache-control'],
+        status: response.status,
+        isInitialHarvestComplete,
+        isOrdersFeed,
+      });
+
+      if (rpdeValidationErrors.length > 0) {
+        if (multibar) multibar.stop();
+        logError(`\nFATAL ERROR: RPDE Validation Error(s) found on RPDE feed ${feedContextIdentifier} page "${url}":\n${rpdeValidationErrors.map(error => `- ${error.message.split('\n')[0]}`).join('\n')}\n`);
+        process.exit(1);
+      }
+
+      context.currentPage = url;
+      if (json.next === url && json.items.length === 0) {
+        if (!isInitialHarvestComplete) {
+          if (context._progressbar) {
+            context._progressbar.update(context.validatedItems, {
+              pages: context.pages,
+              responseTime: Math.round(responseTime),
+              ...progressFromContext(context),
+              status: context.items === 0 ? 'Harvesting Complete (No items to validate)' : 'Harvesting Complete, Validating...',
+            });
+            context._progressbar.setTotal(context.totalItemsQueuedForValidation);
+          }
+          isInitialHarvestComplete = true;
+        }
+        if (WAIT_FOR_HARVEST || VALIDATE_ONLY) {
+          await onFeedEnd();
```

Comment:
Given how WAIT_FOR_HARVEST is described, should we stop harvesting at this point? (and likewise for all other `onFeedEnd()` calls)

Reply:
I'm reluctant to change any harvesting behaviour. Can you think of a better description for it?